RearrangingCars.java

```diff
@@ -0,0 +1,207 @@
+package CodeU_Assignment6;
+
+import java.util.ArrayList;
+import java.util.HashSet;
+
+public class RearrangingCars {
+    /** rearrangingSteps(initial, expected) returns the steps required to arrange the cars
+     *  according to the "expected" rearrangement, starting from the "initial" rearrangement. 
+     *  
+     *  Solution:
+     *      1. Check if the current and the expected rearrangements are valid, if not, then throw an exception.
+     *      2. If the arrays are of length 0 or are exactly equal, there is no steps to be taken, hence return "".
+     *      3. Otherwise, call rearrangingStepsHelper() to get the required series of steps. 
+     *         Return the required steps as a String.           
+     * 
+     * @param initial: an array representing initial rearrangement of the cars
+     * @param expected: an array representing expected rearrangement of the cars
+     * @return String: A string containing a series of steps required to arrange the cars
+     *  according to the "expected" rearrangement, starting from the "initial" rearrangement.
+     * @throws RearrangementException 
+     */
+    public static String rearrangingSteps(int [] initial, int [] expected) throws RearrangementExceptions{

```

Comment:
After working on Miruna's review I've noticed that she destroys `initial` array as she doesn't prepare a copy of it. You do the same thing, I suppose. In my solution I use swapCycles which I thought is a waste of memory. But now I see that It's impossible to solve the task without allocating one more array or destroying initial. Correct me if I'm not right (it's question to all girls :))